From 04defe34008d0f7561c4546829954d2c9d4f2aaf Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 21 Jan 2025 10:42:53 +0200
Subject: [PATCH] [FSW-7256] fs1: recovery pb config

With this patch added support to call api which
configures protection bits for psoc boot security
and psoc pad config pages.

Change-Id: Iea90b609390a00cf134d626d30fb91c14263483a
---

diff --git a/arch/fs1/arc/dev_security.c b/arch/fs1/arc/dev_security.c
index d9e56c0..1e077ad 100644
--- a/arch/fs1/arc/dev_security.c
+++ b/arch/fs1/arc/dev_security.c
@@ -487,7 +487,7 @@
 	regmap_writel(data, spl_base + pb_reg[grp] + (set_idx * sizeof(uint32_t)));
 }
 
-void boot_rom_program_prot_bits(struct pb_page_ctx *pb_cfg_data, uint32_t pb_cfg_size)
+static void boot_rom_program_prot_bits(struct pb_page_ctx *pb_cfg_data, uint32_t pb_cfg_size)
 {
 	uint32_t mask_array[PB_MAX_REG_COPY_PER_GROUP];
 	struct pb_grp_ctx grp_cfg;
@@ -554,3 +554,10 @@
 		} /* group for loop */
 	} /* page for loop */
 }
+
+void security_pb_recovery_config(void)
+{
+	struct pb_page_ctx pb_rec_cfg[] = PROT_BITS_RECOVERY_CFG;
+
+	boot_rom_program_prot_bits(pb_rec_cfg, ARRAY_SIZE(pb_rec_cfg));
+}
diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
index 9c2d483..86fe3c8 100644
--- a/arch/fs1/arc/dev_security.h
+++ b/arch/fs1/arc/dev_security.h
@@ -191,6 +191,7 @@
 	const struct security_rr_ctx *rr_ip_ctx);
 void security_rr_recovery_config(bool enable);
 void security_handle_jtag_if(void);
+void security_pb_recovery_config(void);
 
 extern const uint8_t rom_key_hashes[NUM_KEYS_IN_ROM][SHA384_SUM_LEN];
 
diff --git a/common/recovery.c b/common/recovery.c
index 6929e49..f2f0525 100644
--- a/common/recovery.c
+++ b/common/recovery.c
@@ -27,6 +27,7 @@
 		boot_rom_pcie_link_init();
 
 	security_rr_recovery_config(true);
+	security_pb_recovery_config();
 
 	boot_rom_d2d_link_init();
 
